# New_Pipeline_2023
UROP pipeline based on new taxonomic json
The Zenodo json file: 
https://zenodo.org/badge/latestdoi/664631685


HTML 
<a href="https://zenodo.org/badge/latestdoi/664631685"><img src="https://zenodo.org/badge/664631685.svg" alt="DOI"></a>

Package requirements: 
PySimpleGUI, numpy, pandas, json, time, os, datetime, alive_progress, BioPython, PIL, svglib, rlPyCairo, 
https://github.com/etetoolkit/ete/archive/ete4.zip

Installation instructions

Command prompt in terminal: 

1)          pip install virtualenv
2)          mkdir Annotator
3)          cd Annotator
4)          python3.11 -m venv Annotatorenvironment
5)          cd Annotatorenvironment/Scripts
6)          activate.bat
6)          pip install -r requirements.txt

Before running scripts: 

Create an account with the NCBI - if you would like to construct a phylogenetic tree. 
Follow the URL: https://account.ncbi.nlm.nih.gov/signup/?back_url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2F
Generate an API key by going to account settings --> API Key Management

Usage: 

From GUI: 

Run microbeAnnotatorCaller.py
Follow the prompts provided
Ensure that the directory contains the file colourbar.png if making a tree. 

From command line: 

There are 3 modules: microbiomeAnnotator_condensed, tree_builder, childnodes
Ensure that the correct module is loaded for the usage intended. 

microbiomeAnnotator_condensed usage: 
ann(tax dictionary, input directory, output directory, count, section, casesensitivity)
            tax dictionary = NCBI_tax_dictionary8.json (This loads with the repo from github)
            input directory = Location of your bioc format files to be annotated
            output directory = Folder to save the annotated files to. 
            count = 0 (Used for assigning ids to taxonomic entities)
            section = ALL (ALL by default but allows user to select section of the text to annotate)
            casesensitivity = NO (NO by default but when switched to YES only exact matches are accepted)
tree_builder usage
T( ids found text file, output directory, childnodes directory, input directory)
            ids found text file = (This is automatically generated by the microbiomedAnnotator_condensed module. However, you may generate your own text file by entering each NCBI taxonomic id on a separate line in a text file. There is an example of this is Testset4)
            output directory = Folder to save the trees to. 
            childnodes directory = childnodes.txt (This loads with the repo from github. This module updates this text file)
            input directory = Title of the final tree produced: Location of your bioc format files to be annotated or a manually determined title


childnodes usage
C(childnodes directory, ids found text file, ncbiapikey)
            childnodes directory = childnodes.txt (This loads with the repo from github. This module updates this text file)
            ids found text file = ids.txt (This is automatically generated by the microbiomedAnnotator_condensed module. However, you may generate your own text file by entering each NCBI taxonomic id on a separate line in a text file. There is an example of this is Testset4)
            ncbiapikey = Should be generated from the section above titled 'Before running scripts'